user-interface/src/main/java/life/qbic/datamanager/views/CancelConfirmationNotificationDialog.java

```diff
@@ -0,0 +1,47 @@
+package life.qbic.datamanager.views;
+
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import life.qbic.datamanager.views.notifications.NotificationDialog;
+
+/**
+ * Warns the user that they are about to cancel a dialog and lose input data
+ * <p>
+ * This dialog is to be shown when a dialog is canceled via the Esc key or the Cancel button
+ */
+public class CancelConfirmationNotificationDialog extends NotificationDialog {
+
+  public CancelConfirmationNotificationDialog() {
+    customizeHeader();
+    setCancelable(true);
+    Button redButton = new Button("Discard");
+    redButton.addClassName("danger");
+    setConfirmButton(redButton);
+    Button cancelButton = new Button("Continue");
+    setCancelButton(cancelButton);
+  }
+
+  public CancelConfirmationNotificationDialog withBodyText(String mainText) {
+    content.add(new Span(
+        mainText));
```

Comment:
I'd recommend to call content.removeall() before adding a new span here to keep the behaviour consistent.